__tests__/app.test.js

```diff
@@ -131,38 +131,113 @@ describe("GET /api/articles", () => {
-                    votes: expect.any(Number),
-                    created_at: expect.any(String),
-                    author: expect.any(String),
-                    body: expect.any(String),
-                    article_id: expect.any(Number),
-                })
-            })
-        })
-    })
-    test("comments should be arranged most recent comments first", () => {
-        return request(app).get('/api/articles/1/comments').expect(200).then(({body: {comments}}) => {
-            expect(comments).toBeSortedBy('created_at', { descending: false })
-        })
-    })
-    test("should return error 404 not found when given an invalid article ID", () => {
-        return request(app).get('/api/articles/99999/comments').expect(404).then(({body: {msg}}) => {
-            expect(msg).toEqual('Not Found')
-        })
-    })
-    test("should return error 400 Bad Request when given an invalid article ID data type", () => {
-        return request(app).get('/api/articles/forklift/comments').expect(400).then(({body: {msg}}) => {
-            expect(msg).toEqual('Bad Request')
-        })
-    })
-})
\ No newline at end of file
+  test("returns an array of comments for the given article_id", () => {
+    return request(app)
+      .get("/api/articles/1/comments")
+      .expect(200)
+      .then(({ body: { comments } }) => {
+        expect(comments.length).toBe(11);
+      });
+  });
+  test("comments should be formatted correctly", () => {
+    return request(app)
+      .get("/api/articles/1/comments")
+      .expect(200)
+      .then(({ body: { comments } }) => {
+        comments.forEach((comment) => {
+          expect(comment).toMatchObject({
+            comment_id: expect.any(Number),
+            votes: expect.any(Number),
+            created_at: expect.any(String),
+            author: expect.any(String),
+            body: expect.any(String),
+            article_id: expect.any(Number),
+          });
+          expect(comment.article_id).toBe(1);
+        });
+      });
+  });
+  test("comments should be arranged most recent comments first", () => {
+    return request(app)
+      .get("/api/articles/1/comments")
+      .expect(200)
+      .then(({ body: { comments } }) => {
+        expect(comments).toBeSortedBy("created_at", { descending: false });
+      });
+  });
+  test("should return 200 code and an empty array when there is an article present with no comments", () => {
+    return request(app)
+      .get("/api/articles/2/comments")
+      .expect(200)
+      .then(({ body: { comments } }) => {
+        expect(comments).toEqual([]);
+      });
+  });
+  test("should return error 400 Bad Request when given an invalid article ID", () => {
+    return request(app)
+      .get("/api/articles/forklift/comments")
+      .expect(400)
+      .then(({ body: { msg } }) => {
+        expect(msg).toEqual("Bad Request");
+      });
+  });
+});
+
+describe("POST /api/articles/:article_id/comments", () => {
```

Comment:
This endpoint would benefit from a few more tests, have a go at implementing:

Status 201: Created
    Ignores any unnecessary properties on the request body.

Status 404: Not found, username does not exist
    Responds with error object including status and message
    e.g const newComment = {
      username: "forkliftMaster23",
      body: "Test Comment",
    };
    Remember that any comment will need a valid, and existing user in the database - so a random username like this above should send a 404 back as that user doesn't exist